src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -330,12 +335,27 @@ private boolean checkImmutableIndices(Object request, ActionListener listener) {
         }
         
         if(request instanceof IndexRequest) {
-            if(complianceConfig != null && complianceConfig.isIndexImmutable(request)) {
+            if(complianceConfig != null && isIndexImmutable(request)) {
```

Comment:
move check `complianceConfig != null ` to `isIndexImmutable()`.

Reply:
Done!